### ISSUE TITLE
12$: Improve diagnostic: f = () {}

### ISSUE THREAD

**Issue by strager** [quick-lint/quick-lint-js#461]
```js
    client.on('open', () {
      ws.send('something');
    });
```

Expected: "missing '=>'"

**Comment by AmanGupta03** [comment]
I claim this for-hire task. I expect payment after I complete this task. I will email the quick-lint-js team if I am assigned this task.

**Comment by strager** [comment]
The `looks_like_arrow_function_body` variable in `parse-expression.cpp` looks like a good place to start poking around.

**Comment by HicaroD** [comment]
I claim this for-hire task. I expect payment after I complete this task. I will email the quick-lint-js team if I am assigned this task.
